ciri/api/api.c

```diff
@@ -99,32 +99,39 @@ retcode_t iota_api_get_trytes(iota_api_t const *const api,
                               get_trytes_req_t const *const req,
                               get_trytes_res_t *const res) {
   retcode_t ret = RC_OK;
-  flex_hash_array_t *iter = NULL;
+  hash243_queue_entry_t *iter = NULL;
   flex_trit_t tx_trits[FLEX_TRIT_SIZE_8019];
-  tryte_t tx_trytes[NUM_TRYTES_SERIALIZED_TRANSACTION + 1];
+  trit_array_p tmp_trits = NULL;
+
   DECLARE_PACK_SINGLE_TX(tx, txp, pack);
 
-  if (flex_hash_array_count(req->hashes) > api->conf.max_get_trytes) {
+  if (hash243_queue_count(&req->hashes) > api->conf.max_get_trytes) {
     return RC_API_MAX_GET_TRYTES;
   }
-  LL_FOREACH(req->hashes, iter) {
+  CDL_FOREACH(req->hashes, iter) {
     hash_pack_reset(&pack);
+    // flex_trit to trit_array_p for iota_tangle_transaction_load
+    tmp_trits = trit_array_new_from_flex(iter->hash, NUM_TRITS_HASH,
```

Comment:
we should use a hash243_queue here and not a trit_array?

Reply:
I should put a FIXME here. 
For fixing it, @thibault-martinez could you provide something like `iota_tangle_transaction_load` accept `flex_trits` or `hash243_queue`?

Reply:
I agree that it would be handy and I have thought about it.
The issue is that by accepting `flex_trit_t *` we lose the size attribute hence genericity. 
In the meantime, I really think there's no need for dynamic allocation here.

Reply:
I thought `iota_tangle_transaction_load` will take `flex_trit_t`, so `trit_array_new_from_flex` is a temporary implementation which keeps code clean and can be removed easily. 
If `flex_trit_t` is not a plan for `iota_tangle_transaction_load`, I should change code here.